src/connectionController.ts

```diff
@@ -596,6 +657,11 @@ export default class ConnectionController {
   }
 
   private async _removeSecretsFromKeychain(connectionId: string) {
+    await this._storageController.deleteSecret(connectionId);
+    // Even though we migrated to SecretStorage from keytar, we are still removing
+    // secrets from keytar to make sure that we don't leave any left-overs when a
+    // connection is removed. This block can safely be removed after our migration
+    // has been out for some time.
     if (ext.keytarModule) {
       await ext.keytarModule.deletePassword(this._serviceName, connectionId);
```

Comment:
Does this error if the password doesn't exist? If so maybe it's worth `try` `catch`ing as we could have already deleted it in the migration.

Reply:
Yea I think that is worth adding. On it.

Reply:
Addressed this in the [following commit](https://github.com/mongodb-js/vscode/pull/552/commits/808e60e8b6d1b60ee5d71103c2ff858bb8baaa7d)